connect/src/com/telenor/connect/SdkProfile.java

```diff
@@ -24,10 +24,23 @@
     WellKnownAPI.WellKnownConfig getWellKnownConfig();
     boolean isInitialized();
 
-    void onStartAuthorization(OnStartAuthorizationCallback callback);
+    void initializeFromUri(
+            Map<String, String> parameters,
+            List<String> uiLocales,
+            Uri initFrom,
+            OnDeliverAuthorizationUriCallback callback);
 
-    interface OnStartAuthorizationCallback {
-        void onSuccess();
+    void onStartAuthorization(
+            Map<String, String> parameters,
+            List<String> uiLocales,
+            OnStartAuthorizationCallback callback);
+
+    interface OnDeliverAuthorizationUriCallback {
+        void onSuccess(Uri authorizationStartUri);
         void onError();
     }
+
+    interface OnStartAuthorizationCallback extends OnDeliverAuthorizationUriCallback {
```

Comment:
1. I'm not sure I understand this argument. Putting an activity in a manifest, shares this with the OS, and does not expose developers to it more than it is currently. Also the encapsulation you did, makes the code a bit harder to follow.
2. "Connect Id could have used it for something else" like what? I think this shouldOverrideUrlLoading is the wrong hook to achieve what you want.